src/main/js/controls/Graph/Graph.js

```diff
@@ -69,12 +69,20 @@ const setCanvasWidth = (container, config) => {
  * @returns {undefined} - returns nothing
  */
 const setCanvasHeight = (config) => {
-    config.canvasHeight =
-        getYAxisHeight(config) +
-        (config.padding.bottom +
+    if (config.showLabel || config.axis.x.show) {
+        config.canvasHeight =
+            getYAxisHeight(config) +
+            (config.padding.bottom +
+                config.padding.top +
+                config.axisLabelHeights.x) *
+                2;
+    } else {
+        config.canvasHeight =
+            getYAxisHeight(config) +
+            config.padding.bottom +
             config.padding.top +
-            config.axisLabelHeights.x) *
-            2;
+            config.axisLabelHeights.x;
```

Comment:
Why we are removing multiplication here?

Reply:
We were multiplying padding-top, padding-bottom, and x-axis label height by 2 so that we can get enough space for the legend or otherwise it will be overlapping the x-axis label or its values. If suppose, some consumers do not use either x-axis values or x-axis label by setting show: false for x-axis or showLabel: false respectively, then we do not need the multiplication part, since there will be no case of legend overlapping on the x-axis label or its values.

Hence I am checking [this condition](https://github.com/cerner/carbon-graphs/pull/213/commits/616164b66e7a4f0ba408397f05c00fa7bf886de0#diff-4c7acbc816e711a56dc8fffdd18a4149R72) and if it satisfies, then we do the multiplication part. Else if both x-axis values and x-axis label are set to false, we are removing the multiplication part since its no longer necessary in this particular case. This way, if there is a second canvas, it will display after a space/padding of 10px from the first graph if x-axis values and x-axis label are set to false and bindLegendToId is provided.

Reply:
Thanks for the explanation.

- Did we have already have test cases to cover both scenarios?

- Would it be a good idea to add both scenarios in dev examples?

Reply:
I have added the test cases to cover the scenarios [here](https://github.com/JananiGunasekaran/carbon-graphs/commit/c9ea2fb27af1dc46e736037ed711fa2089ef0a20).

I have not added this scenario in dev example since it would not be easy to visualise this change on the UI. We can get to know that the spacing has reduced only if we select the element and see its size like this: ( When both x-axis show and showLabel are set to false)

![image](https://user-images.githubusercontent.com/60426528/82549604-ce704100-9b7a-11ea-8b18-d3f6bb7bd8c7.png)

When both x-axis show and showLabel are set to true, we can see the spacing for the label and x-axis values when we select the element :

![image](https://user-images.githubusercontent.com/60426528/82549726-037c9380-9b7b-11ea-8f8e-37ac7e4f0945.png)

This is the reason why i have not added this scenario in dev example - it would not be easy to visualise this change on the UI.